src/ZstdCompressor.hpp

```diff
@@ -23,6 +23,7 @@
 #include "Compressor.hpp"
 #include "NonCopyable.hpp"
 
+#include <algorithm>
```

Comment:
Suggestion: Move this to the `.cpp` file instead since it's not required by anything in the header file.